osu.Game.Rulesets.Osu/UI/OsuPlayfield.cs

```diff
@@ -54,7 +54,8 @@ public OsuPlayfield()
                 approachCircles = new ProxyContainer { RelativeSizeAxes = Axes.Both },
             };
 
-            hitPolicy = new OrderedHitPolicy(HitObjectContainer);
+            hitPolicy = new StartTimeOrderedHitPolicy();
+            hitPolicy.SetHitObjects(HitObjectContainer.AliveObjects);
```

Comment:
What was the reason behind this API choice of passing down an enumerable instead of the entire hit object container as before? I'm not sure about it, because what this line ends up doing is making the hit policy store a live ordered enumerable (backed by a list, but still a single enumerable), which seems slightly dodge.

Reply:
I considered that but sending drawable components (and containers nonetheless) down is very weird to me. There's a case in mania where I've sent down the HOC, and I'm not happy with that either.

It's not a super big deal if you + @peppy think it's better to send the HOC down instead.

Reply:
I'd prefer we don't pass live enumerables between objects like this (and if we are, it needs to be explicitly documented). We do this elsewhere with the same property (`AliveObjects`) and it has led to bugs and performance issues in the editor in the past.

How about making an interface and passing via that, rather than as the drawable type?

Reply:
So something like the following?
```
interface IHitObjectContainer
{
    IEnumerable<DrawableHitObject> AliveObjects { get; }
}
```

Reply:
Yeah, seems fine. I'm still not sure about exposing things in interfaces as `IEnumerable` (what does microsoft do? it seems like a bad idea..)

Reply:
The only example I can come up with from Microsoft is Roslyn, which exposes [`Compilation.SyntaxTrees`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.compilation.syntaxtrees?view=roslyn-dotnet#Microsoft_CodeAnalysis_Compilation_SyntaxTrees) and a few other things like projects from solutions as `IEnumerable`s. But a lot of other stuff is also `ImmutableArray`, which unfortunately we can't really use here since these are dynamic things.